core/deployment/src/main/java/io/quarkus/deployment/builditem/RawCommandLineArgumentsBuildItem.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.deployment.builditem;
+
+import java.util.function.Supplier;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+import io.quarkus.deployment.recording.BytecodeRecorderImpl;
+import io.quarkus.runtime.StartupContext;
+
+/**
+ * A build item that represents the raw command line arguments as they were passed to the application.
+ *
+ * No filtering is done on these parameters.
```

Comment:
Maybe you could add a comment about this proxy magic? It's not the typical build item.